_source/_posts/2019-09-26-getting-started-with-elk.md

```diff
@@ -0,0 +1,375 @@
+
+JHipster applications require the specific user roles **ROLE_USER** and **ROLE_ADMIN** as claims in the ID Token. In the Okta Developer Console, go to **Users > Groups** and create a group for each JHipster role, and add users to each group.
+
+Now go to **API > Authorization Servers**, select the **default** server and **Add Claim** with the following settings:
+1. Name: groups
+2. Include in token type: ID Token, Always
+3. Value type: Groups
+4. Filter: Matches regex, set the Regex **.\***
+
+### Enable Debug Logs and Zipkin
+
+To send debug logs to the JHipster Console, let's update the log level in the prod profile.  Edit **src/main/resources/config/application-prod.yml** to set the level for each service (`blog-app`, `store-app` and `gateway-app`) to  **DEBUG** for the **com.okta.developer.\*** logger. For example, in the blog application-prod.yml:
+
+```yaml
+logging:
+    level:
+        com.okta.developer.blog: DEBUG
+```
+
+Also, for each service, update the **LoggingAspectConfiguration** to load when the prod profile is active. Change the `@Profile` annotation:
+
+```Java
+@Configuration
+@EnableAspectJAutoProxy
+public class LoggingAspectConfiguration {
+
+    @Bean
+    @Profile({JHipsterConstants.SPRING_PROFILE_DEVELOPMENT, JHipsterConstants.SPRING_PROFILE_PRODUCTION})
+    public LoggingAspect loggingAspect(Environment env) {
+        return new LoggingAspect(env);
+    }
+}
+```
+[**Zipkin**](https://zipkin.io/) is a distributed tracing system that helps troubleshooting latency issues in microservices architectures. With a traceId propagated from service to service, calls to different services can be correlated and analyzed as a part of the same flow. The Zipkin server and UI are provided with the JHipster Console, and JHipster apps can integrate with Zipkin through [**Spring Cloud Sleuth**](https://cloud.spring.io/spring-cloud-sleuth/reference/html/#sleuth-with-zipkin-via-http). 
+To enable Zipkin tracing, add the zipkin profile to `blog-app`, `gateway-app` and `store-app` in **docker-compose/docker-compose.yml**.
+
+```yaml
+- SPRING_PROFILES_ACTIVE=prod,swagger,zipkin
+```
+You also need to rebuild the docker images with the **zipkin** profile, for `blog-app`, `store-app` and `gateway-app` with the following Maven command:
+```SHELL
+./mvnw package -Pprod -Pzipkin verify jib:dockerBuild -DskipTests
+```
+
+>ProTip:  If you're on a system with bash shell, like Linux or MacOs, you can do this from the `jhipster` folder to build each project at once:
+```
+for i in blog gateway store
+do 
+cd $i 
+./mvnw package -Pprod -Pzipkin verify jib:dockerBuild -DskipTests
+cd ..
+done
+```
+
+### Run the Monitored Microservices Architecture
+
+Are you ready for the best? Go to the `docker-compose` folder and start the services with the following command:
+
+```SHELL
+docker-compose up
+```
+
+You will see a huge amount of logging while each service starts.
```

Comment:
```suggestion
You will see a lot of logging while each service starts.
```

Reply:
this seems more like a personal style thing. imma leave it.